src/NUnitConsole/nunit3-console/ConsoleRunner.cs

```diff
@@ -433,11 +435,51 @@ public static TestPackage MakeTestPackage(ConsoleOptions options)
                 package.AddSetting(FrameworkPackageSettings.DefaultTestNamePattern, options.DefaultTestNamePattern);
 
             if (options.TestParameters != null)
-                package.AddSetting(FrameworkPackageSettings.TestParameters, options.TestParameters);
+            {
+                var dictionary = new Dictionary<string, string>();
+
+                // Exactly what the framework used to do.
+                // Now can be independently changed without breaking backwards compatibility with old frameworks.
+                foreach (var param in options.TestParameters.Split(';'))
```

Comment:
Use a review comment (like this one) if you want to tell the reviewers why you did something. The comment in the code will only have meaning for us while this change is in progress but will just be confusing afterward. Anyway, we understand what you are doing. :-)

That said, It seems to me to make more sense for the ConsoleOptions class to create the dictionary in the first place, rather than making the runner do it. @oznetmaster What do you think?